packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -610,6 +716,66 @@ export abstract class BaseService extends Resource
     this.node.defaultChild = this.resource;
   }
 
+  /**
+   *   Enable Deployment Alarms which take advantage of arbitrary alarms and configure them after service initialization
+  */
+  public enableDeploymentAlarms(alarmConfig: DeploymentAlarmConfig) {
+    // Throw an error if deployment alarms are already configured
+    if (this.deploymentAlarms) {
+      throw new Error('Deployment alarms are already configured.');
+    }
+    // Throw an error if alarms array is empty
+    if (alarmConfig.alarms.length === 0) {
+      throw new Error('Specify at least one deployment alarm');
+    }
+    this.deploymentAlarms = {
+      alarmNames: alarmConfig.alarms.map(alarm => alarm.alarmName),
+      enable: true,
+      rollback: alarmConfig.behavior !== AlarmBehavior.FAIL_ON_ALARM,
+    };
+  }
+
+  /**
+   *   Disassociate existing deployment alarms
+  */
+  public disableDeploymentAlarms() {
+    if (this.deploymentAlarms) {
+      this.deploymentAlarms = {
+        alarmNames: [],
+        rollback: false,
+        enable: false,
+      };
+    }
+  }
+
+  /**
+   *   Add an alarm based on ECS metric which appended to the list of deployment alarms at synthesis time
+   *   Customize the alarm threshold and evaluationPeriods using 2nd argument ecsMetricAlarmProps.
+   *   `threshold  value 85`
+   *   `evaluationPeriods value 3`
+  */
+  public createEcsMetricAlarm(metric: EcsMetric, ecsMetricAlarmProps?: EcsMetricAlarmProps): cloudwatch.Alarm {
+
+    const ecsMetric = this.metric(metric, ecsMetricAlarmProps?.metricProps);
+    const alarmName = Names.uniqueId(this);
+    const metricAlarm = new cloudwatch.Alarm(this, alarmName,
+      ecsMetricAlarmProps?.useAsDeploymentAlarm && ecsMetricAlarmProps?.alarmProps ? ecsMetricAlarmProps.alarmProps : {
+        metric: ecsMetric,
+        threshold: ecsMetricAlarmProps?.alarmProps?.threshold || 85,
+        evaluationPeriods: ecsMetricAlarmProps?.alarmProps?.evaluationPeriods || 3,
+        alarmName,
+      });
```

Comment:
Hi - I might be missing the context here - looks like if I set `useAsDeploymentAlarm` to `false` - this code would still create a default alarm for me? (also it's using values from `ecsMetricAlarmProps.alarmProps` here

could you explain what's the use case for `useAsDeploymentAlarm === false`

Reply:
I made changes as follows so that if we set  useAsDeploymentAlarm === false it wont create deployment alarms for us.
Please check the latest commit for updated changes

```
if (ecsMetricAlarmProps?.useAsDeploymentAlarm) {
      if (!this.deploymentAlarms) {
        this.deploymentAlarms = {
          enable: true,
          rollback: true,
          alarmNames: [],
        };
      }
      this.deploymentAlarms.alarmNames.push(metricAlarm.node.id);
```